tests/components/bang_olufsen/test_media_player.py

```diff
@@ -129,13 +126,35 @@ async def test_async_update_sources_outdated_api(
     )
 
 
+async def test_async_update_sources_remote(
+    hass: HomeAssistant, mock_mozart_client, mock_config_entry
+) -> None:
+    """Test _async_update_sources is called when there are new video sources."""
+
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    websocket = hass.data[DOMAIN][mock_config_entry.entry_id].websocket
```

Comment:
We should not touch internals in tests. Rather use the mock to fetch the callback function, an example can be found in knocki tests

Reply:
@joostlek is right. Just in case it's not clear, you should get the callback functions to call from the mocked calls which register them to `MozartClient.get_notification_notifications`, `MozartClient.get_on_connection_lost` and so on.

There's something similar also in `cast` tests: https://github.com/emontnemery/home-assistant/blob/378310c86de52b71fd86c39e6d182eb3205d2cd3/tests/components/cast/test_media_player.py#L215

Reply:
Thank you two for the suggestions, I think I've got it now.